swxsoc/swxdata.py

```diff
@@ -0,0 +1,841 @@
+        """
+        return self._spectra
+
+    @property
+    def data(self):
+        """
+        (`dict`) A `dict` containing each of `timeseries` and `support`.
+        """
+        return {
+            "timeseries": self.timeseries,
+            "spectra": self.spectra,
+            "support": self.support,
+        }
+
+    @property
+    def meta(self):
+        """
+        (`collections.OrderedDict`) Global metadata associated with the measurement data.
+        """
+        return self._timeseries.meta
+
+    @property
+    def time(self):
+        """
+        (`astropy.time.Time`) The times of the measurements.
+        """
+        t = Time(self._timeseries.time)
+        # Set time format to enable plotting with astropy.visualisation.time_support()
+        t.format = "iso"
+        return t
+
+    @property
+    def time_range(self):
+        """
+        (`tuple`) The start and end times of the times.
+        """
+        return (self._timeseries.time.min(), self._timeseries.time.max())
+
+    def __repr__(self):
+        """
+        Returns a representation of the `SWXData` class.
+        """
+        return self.__str__()
+
+    def __str__(self):
+        """
+        Returns a string representation of the `SWXData` class.
+        """
+        str_repr = f"SWXData() Object:\n"
+        # Global Attributes/Metedata
+        str_repr += f"Global Attrs:\n"
+        for attr_name, attr_value in self._timeseries.meta.items():
+            str_repr += f"\t{attr_name}: {attr_value}\n"
+        # TimeSeries Data
+        str_repr += f"TimeSeries Data:\n"
+        for var_name in self._timeseries.colnames:
+            str_repr += f"\t{var_name}\n"
+        # Support Data
+        str_repr += f"Support Data:\n"
+        for var_name in self._support.keys():
+            str_repr += f"\t{var_name}\n"
+        # Spectra Data
+        str_repr += f"Spectra Data:\n"
+        for var_name in self._spectra.keys():
+            str_repr += f"\t{var_name}\n"
+        return str_repr
+
+    @staticmethod
+    def global_attribute_template(
+        instr_name: str = "", data_level: str = "", version: str = ""
+    ) -> OrderedDict:
+        """
+        Function to generate a template of the required ISTP-compliant global attributes.
+
+        Parameters
+        ----------
+        instr_name : `str`
+            The instrument name. Must be "eea", "nemisis", "merit" or "spani".
+        data_level : `str`
+            The data level of the data. Must be "l0", "l1", "ql", "l2", "l3", "l4"
+        version : `str`
+            Must be of the form X.Y.Z.
+
+        Returns
+        -------
+        template : `collections.OrderedDict`
+            A template for required global attributes.
+        """
+        meta = SWXSchema().global_attribute_template()
+
+        # Check the Optional Instrument Name
+        if instr_name:
+            if instr_name not in swxsoc.INST_NAMES:
```

Comment:
This is where we might need a configuration file or input that will determine the valid instrument names.